File: Frontend/src/app/classes/pagination/paged-list.ts
export class PagedList<T> {
    currentPage: number = 0;
    pageCount: number = 0;
    pageSize: number = 0;
    recordCount: number = 0;
    records: T[] = [];

    constructor(pageInfo?) {
        if (!pageInfo) return;
        if (pageInfo.currentPage) this.currentPage = pageInfo.currentPage;
        if (pageInfo.pageCount) this.pageCount = pageInfo.pageCount;
        if (pageInfo.pageSize) this.pageSize = pageInfo.pageSize;
        if (pageInfo.recordCount) this.recordCount = pageInfo.recordCount;
    }
}

File: Backend/Topsys.TopConWeb.SharedKernel/Common/PagedList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Topsys.TopConWeb.SharedKernel.Common
{
    public class PagedList<T>
    {
        public int CurrentPage { get; set; }

        public int PageCount { get; set; }

        public int PageSize { get; set; }

        public int RecordCount { get; set; }

        public IEnumerable<T> Records { get; set; }

        public string Info {
            get
            {
                return "{"
                    + "\"currentPage\": " + CurrentPage 
                    + ", \"pageCount\": " + PageCount 
                    + ", \"pageSize\": " + PageSize 
                    + ", \"recordCount\": " + RecordCount
                    + "}";
            }
        }

    }
}
